File: scala/examples/src/test/scala/stripe/ProcessorTest.scala
package stripe

import org.scalatest._
import org.scalatest.flatspec._
import org.scalatest.matchers.should.Matchers


class  ProcessorTest extends AnyFlatSpec with Matchers {

  val processor = new Processor

  "A processor" should "test 1" in {

    val acceptedLanguages = "en-US, fr-CA, fr-FR"
    val supportedLanguages = List("fr-FR","en-US")

    val result = processor.process(acceptedLanguages, supportedLanguages)

    result.size should be (2)
    result(0) should be ("en-US")
    result(1) should be ("fr-FR")

  }

  "A processor" should "test 2" in {

    val acceptedLanguages = "en-US, fr-CA, fr-FR,fr-CA"
    val supportedLanguages = List("fr-CA","en-US")

    val result = processor.process(acceptedLanguages, supportedLanguages)

    result.size should be (2)
    result(0) should be ("en-US")
    result(1) should be ("fr-CA")

  }


  "A processor" should "test 3" in {

    val acceptedLanguages = "     "
    val supportedLanguages = List("fr-CA","en-US")

    val result = processor.process(acceptedLanguages, supportedLanguages)

    result.size should be (0)

  }


  "A processor" should "test 4" in {

    val acceptedLanguages = "en-US, fr-CA, fr-FR"
    val supportedLanguages = List()

    val result = processor.process(acceptedLanguages, supportedLanguages)

    result.size should be (0)

  }

}

File: prep/interview-strip/src/test/java/ProcessorTest.java
import org.junit.Assert;
import org.junit.Assert.*;
import org.junit.Test;


//import static org.junit.matchers.JUnitMatchers.*;

import java.util.ArrayList;
import java.util.List;

public class ProcessorTest {

    private Processor processor = new Processor();

    @Test
    public void test1(){
        String acceptedLanguages = "en-US, fr-CA, fr-FR";
        List<String> supportedLanguages = List.of("fr-FR", "en-US");

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals("en-US",response.get(0));
        Assert.assertEquals("fr-FR",response.get(1));
        Assert.assertEquals(2,response.size());
    }

    @Test
    public void test2(){
        String acceptedLanguages = "fr-CA, fr-FR";
        List<String> supportedLanguages = List.of("en-US", "fr-FR");

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals("fr-FR",response.get(0));
        Assert.assertEquals(1,response.size());

    }

    @Test
    public void test3(){
        String acceptedLanguages = "en-US";
        List<String> supportedLanguages = List.of("en-US", "fr-CA");

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals("en-US",response.get(0));
        Assert.assertEquals(1,response.size());

    }

    @Test
    public void test4Empty(){
        String acceptedLanguages = "     ";
        List<String> supportedLanguages = List.of("en-US", "fr-CA");

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals(0,response.size());

    }

    @Test
    public void test5Empty(){

        String acceptedLanguages = "     ";
        List<String> supportedLanguages = new ArrayList<>();

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals(0,response.size());

    }

    @Test
    public void test6Empty(){

        String acceptedLanguages = "en-US, fr-CA, fr-FR";
        List<String> supportedLanguages = new ArrayList<>();

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals(0,response.size());

    }

    @Test
    public void test6Duplicates(){
        String acceptedLanguages = "en-US, fr-CA, fr-FR, en-US";
        List<String> supportedLanguages = List.of("fr-FR", "en-US");

        List<String> response = processor.process(acceptedLanguages,supportedLanguages);

        Assert.assertEquals("en-US",response.get(0));
        Assert.assertEquals("fr-FR",response.get(1));
        Assert.assertEquals(2,response.size());
    }

}
